test/test.js

```diff
@@ -926,5 +927,107 @@ describe( 'rollup-plugin-node-resolve', function () {
 				'array-index'
 			]);
 			delete global.sideEffects;
-		}));
+		})
+	);
+
+	describe.only('getPackageInfoForId', () => {
+		it('populates info for main', () => {
+			const resolve = nodeResolve({
+				mainFields: ['main']
+			});
+
+			let entriesInfo;
+
+			return rollup.rollup({
+				input: 'samples/prefer-main/main.js',
+				plugins: [
+					resolve,
+					{
+						transform (code, id) {
+							if (!id.match(/main-entry.js$/)) return;
+							entriesInfo = resolve.getPackageInfoForId(id);
+							return code;
+						}
+					}
+				]
+			}).then(() => {
+				const entriesPkgJsonPath = path.resolve(__dirname, './node_modules/entries/package.json');
+				const root = path.dirname(entriesPkgJsonPath);
+				assert.equal(entriesInfo.browserMappedMain, false);
```

Comment:
Couldn't this become one big `deepEqual` (or `deepStrictEqual` as I hear that the non-strict ones are regarded deprecated)?